### PR TITLE
[JBIDE-20099] Add pod and build log retrievel

### PR DESCRIPTION
**Pull request by jcantrill**
@fbricon @adietish please review

### CODE REVIEW

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/PodLogsJob.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+		}
+		
+		public void stop() {
+			this.running = false;
+			capability.stop();
+		}
+		
+		@Override
+		public void run() {
+			ConsoleUtils.registerConsoleListener(new ConsoleListener(this));
```
Comment (fbricon) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/PodLogsJob.java]: shouldn't you unregister the listener when everything's done?

Comment (jcantrill) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/PodLogsJob.java]: See: https://github.com/jbosstools/jbosstools-openshift/pull/733/files#diff-6badc0be45456741ad5c2f6323526e3fR147

Comment (fbricon) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/PodLogsJob.java]: but would do you get EOF when you've reached the end of the current log? eg. inactive EAP server. Should resume streaming once it receives more hits

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/PodLogsHandler.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+				MessageDialog.openError(HandlerUtil.getActiveShell(event), "No pod selection", "Unable to determine the build pod inorder to retrieve its log.");
+				return;
+			}
+		}
+		showLogs(pod, event);
+	}
+	
+	private void showLogs(IPod pod, ExecutionEvent event) {
+		if(pod == null){
+			MessageDialog.openError(HandlerUtil.getActiveShell(event), "No pod selection", "No pod or build was selected to retrieve a log.");
```
Comment (fbricon) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/PodLogsHandler.java]: No pod selected

Comment (fbricon) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/PodLogsHandler.java]: No pod was selected to retrieve a log

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/PodLogsHandler.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	private static final String BUILDNAME_ANNOTATION = "openshift.io/build.name";
+
+	@Override
+	protected void handleEvent(ExecutionEvent event){
+		ISelection selection = HandlerUtil.getCurrentSelection(event);
+		IPod pod = UIUtils.getFirstElement(selection, IPod.class);
+		if(pod == null) {
+			pod = getPodFromBuild(selection);
+			if(pod == null) {
+				MessageDialog.openError(HandlerUtil.getActiveShell(event), "No pod selection", "Unable to determine the build pod inorder to retrieve its log.");
```
Comment (fbricon) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/PodLogsHandler.java]: No pod selected

Comment (fbricon) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/PodLogsHandler.java]: Unable to determine the build pod in order to retrieve its log

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/PodLogsHandler.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+		showLogs(pod, event);
+	}
+	
+	private void showLogs(IPod pod, ExecutionEvent event) {
+		if(pod == null){
+			MessageDialog.openError(HandlerUtil.getActiveShell(event), "No pod selection", "No pod or build was selected to retrieve a log.");
+			return;
+		}
+		if(!ArrayUtils.contains(STATES, pod.getStatus())) {
+			MessageDialog.openError(HandlerUtil.getActiveShell(event), "Logs Unavailable", NLS.bind("The log is unavailable while the pod is in {0}.  Try again later.", pod.getStatus()));
```
Comment (fbricon) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/PodLogsHandler.java]: The log is unavailable while the pod is in '{0}' state

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/PodLogsJob.java (hunk 1 above)
Comment (jcantrill) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/PodLogsJob.java]: I think we are at the mercy of what the binary does, and probably is contingent on whether the pod is running.  My brief testing showed that you get a EOF on a finished build and when the build completes.  I presume this is a stopped or stopping pod.  If you have an inactive EAP server, presumably the pod is still running, its just not logging to the stream even though the stream is available.  Also, the listener does some cleanup stuff that we only want when the user clicks the red square which is basically remove the console from view.

**Comment by jcantrill** [comment]
@fbricon updated the messages to the user but made no changes to the registration of the console listener

**Comment by fbricon** [comment]
+1 to apply. Works great for me

**Pull request merged**